internal/services/network/virtual_network_resource.go

```diff
@@ -599,6 +599,16 @@ func resourceVirtualNetworkUpdate(d *pluginsdk.ResourceData, meta interface{}) e
 		defer locks.UnlockMultipleByName(routeTables, routeTableResourceName)
 	}
 
+	if payload.Properties.Subnets != nil {
+		// remove readonly properties as they are not managed by TF - large networks can cause ARM API limit errors
+		for i := range *payload.Properties.Subnets {
+			if (*payload.Properties.Subnets)[i].Properties != nil {
+				(*payload.Properties.Subnets)[i].Properties.IPConfigurations = nil
+				(*payload.Properties.Subnets)[i].Properties.PrivateEndpoints = nil
+			}
+		}
+	}
```

Comment:
`payload.Properties` _could_ be `nil` here, causing a crash. Can we move this reset of the R/O properties to inside the correctly nil-checked `if payload.Properties != nil && payload.Properties.Subnets != nil {` below so we don't need to perform the same check twice?